core-blocks/playlist/edit.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	FormFileUpload,
+	IconButton,
+	Placeholder,
+	ServerSideRender,
+	Toolbar,
+	CheckboxControl,
+	PanelBody,
+	SelectControl,
+} from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	InspectorControls,
+	MediaUpload,
+	BlockControls,
+	editorMediaUpload,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+
+class PlaylistEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.initializePlaylist = this.initializePlaylist.bind( this );
+		this.uploadFromFiles = this.uploadFromFiles.bind( this );
+		this.addFiles = this.addFiles.bind( this );
+
+		// check for if ids is set to determine edit state
+		this.state = {
+			editing: ! this.props.attributes.ids,
+		};
+	}
+
+	initializePlaylist() {
+		window.wp.playlist.initialize();
+	}
+
+	uploadFromFiles( event ) {
+		this.addFiles( event.target.files );
+	}
+
+	addFiles( files ) {
+		const { setAttributes } = this.props;
+		editorMediaUpload( {
+			allowedType: [ 'audio', 'video' ],
+			filesList: files,
+			onFileChange: ( media ) => {
+				if ( media && media[ 0 ].mime_type ) {
+					const type = media[ 0 ].mime_type.split( '/' )[ 0 ];
+					const ids = JSON.stringify( media.map( ( item ) => item.id ) );
+					setAttributes( { ids, type } );
+					this.setState( { editing: false } );
+				}
+			},
+		} );
+	}
+
+	toggleAttribute( attribute ) {
+		return ( newValue ) => {
+			this.props.setAttributes( { [ attribute ]: newValue } );
+		};
+	}
+
+	render() {
+		const { attributes, setAttributes, className } = this.props;
+		const { editing } = this.state;
+		const { tracklist, artists, images, style, type } = attributes;
+		const styleOptions = [
+			{ value: 'light', label: __( 'Light' ) },
+			{ value: 'dark', label: __( 'Dark' ) },
+		];
+
+		const onSelectMedia = ( media ) => {
+			//check if there are returned media items and set attributes when there are
+			if ( media && media[ 0 ].url ) {
+				const ids = JSON.stringify( media.map( ( item ) => item.id ) );
+				setAttributes( { tracklist, artists, images, ids, type: media[ 0 ].type } );
+				this.setState( { editing: false } );
+			}
+		};
+
+		const mediaIds = this.props.attributes.ids && this.props.attributes.ids.replace( /^\[(.+)\]$/, '$1' ).split( ',' );
+
+		if ( editing ) {
+			return (
+				<Placeholder
+					icon="media-audio"
+					label={ __( 'Audio/Video Playlist' ) }
+					instructions={ __( 'Select audio or video files from your library, or upload a new ones' ) }
+					className={ className }>
+					<FormFileUpload
+						isLarge
+						multiple
+						className="wp-block-playlist__upload-button"
+						onChange={ this.uploadFromFiles }
+						accept="audiovideo"
+					>
+						{ __( 'Upload' ) }
+					</FormFileUpload>
+					<MediaUpload
+						onSelect={ onSelectMedia }
+						type="audio"
```

Comment:
When you click the Media Library button we need a default type to set for when the modal is opened. I set this to `audio` just because it was the first option of the two. I have no data to support this, but my gut tells me playlists are used more often for Audio than Video so the default might be fair. Thoughts?

Reply:
yeah that makes much more sense. fixed